reanahub/reana-server
# api: improve api rate limiter error messages

closes https://github.com/reanahub/reana-server/issues/455
closes https://github.com/reanahub/reana-ui/issues/241

- created new `handle_rate_limit_error()` error handler to return more verbose error messages in case rate limiter returns an error.
- moved `handle_validation_error()` from a blueprint to app level to handle these kind of errors for other endpoints as well.
- added `JSONDecodeError` exception handling for `launch` endpoint to improve the error message for the invalid format of workflow `parameters`

To test:
- modify rate limiter values for `REANA_RATELIMIT_AUTHENTICATED_USER` and/or `REANA_RATELIMIT_SLOW` to something more strict like: `1/30 second`
- observe that the error messages are now more verbose - `Too many requests in a given amount of time. Only "1 per 5 second" allowed.`
- check other endpoints (e.g. `get_workflows()`) to also use `handle_validation_error()` for handling error messages
- check https://github.com/reanahub/reana-ui/issues/241#issuecomment-1067800198 case with wrong `parameters` format to return more verbose error message
